### PR TITLE
mind cursor column, add audio cue debounce position change setting

### LINKED ISSUE #172523: [Accessibility]: Audio Cues are notplayed a against swift arrow navigation
CC @meganrogge 

* Notes: This is not a regression.

1. Create test.ts:

``` ts
// line 1 comment
xxxx // line 2: intentional error
// line 3 comment
```

1. With screen reader, do the following:

* Move l1 -> l2 (with one second interval): you would hear the audio cue.

* In l2, do round trip quicly between l2 and l1 (i.e., quickly move up to l1 and come back to l2 again with about .5 second interval): You won't hear audio cue.

* Move l3 -> l2 (with one second interval): you would hear the audio cue.

* In l2, do round trip quicly between l2 and l3 (i.e., quickly move down to l3 and come back to l2 again with about .5 second interval): You won't hear audio cue.

## Expected Behavior

Audio cues need to be played consistently just like the one in diff editor.

## System Info

Version: 1.75.0-insider (system setup)
Commit: 07d6f5b35fac559504dec65e3073d796b2d25a55
Date: 2023-01-26T05:23:05.014Z
Electron: 19.1.9
Chromium: 102.0.5005.194
Node.js: 16.14.2
V8: 10.2.154.23-electron.0
OS: Windows_NT x64 10.0.22621
Sandboxed: Yes

### LINKED ISSUE #172525: [Accessibility] Error audio cues are not played on a character level.
CC @meganrogge 

* Notes: This is not a regression.

1. Create test.ts:

``` ts
// line 1 comment
xxxx // line 2: intentional error
// line 3 comment
```

1. With screen reader, do the following:

* In l2, use LeftArrow and RightArrow to navigate the line character by character: You only hear the error audio cue just one time in the first character "x" not in other characters causing the error.

## Expected Behavior

See how the audio cues in diff editor behaves. That's ideal! In diff editor audio cues are played even on a character-level when users use Left/RightArrow keys. This is fantastic because users can hear where the diff starts and ends precisely. However, the error/warning audio cues are not working like that.

## System Info

Version: 1.75.0-insider (system setup)
Commit: 07d6f5b35fac559504dec65e3073d796b2d25a55
Date: 2023-01-26T05:23:05.014Z
Electron: 19.1.9
Chromium: 102.0.5005.194
Node.js: 16.14.2
V8: 10.2.154.23-electron.0
OS: Windows_NT x64 10.0.22621
Sandboxed: Yes

### PR DESCRIPTION
**Pull request by meganrogge**
fix #172525
fix #172523

### CODE REVIEW

**Comment by hediet** [comment]
The fix for #172525 would be to adjust the debounce mechanism.

No *line* feature depends on the column of the cursor, so watching column changes shouldn't do anything, except working around any debounce.

**Review by hediet — CHANGES_REQUESTED** [review]
This also needs to be adjusted:

https://github.com/microsoft/vscode/pull/173334/files#diff-c38e86f6f2004946c34efd01d7d3298d926dc5b266b4a471c4a47aaedfe4be15R157-R158

File: src/vs/workbench/contrib/audioCues/browser/audioCueLineFeatureContribution.ts [hunk 1]
```diff
@@ -191,14 +193,14 @@ class MarkerLineFeature implements LineFeature {
...
+				isPresent: (position) => {
 					const hasMarker = this.markerService
 						.read({ resource: model.uri })
 						.some(
 							(m) =>
 								m.severity === this.severity &&
-								m.startLineNumber <= lineNumber &&
-								lineNumber <= m.endLineNumber
+								m.startLineNumber <= position.lineNumber &&
+								position.lineNumber <= m.endLineNumber
```
[the remark this answers is not in the document]
Comment (meganrogge) [reply src/vs/workbench/contrib/audioCues/browser/audioCueLineFeatureContribution.ts:203]: when I tested the changes, I was surprised to see that they fixed this without having to account for `position.column`. 🤔 any idea why that would be?

**Comment by meganrogge** [comment]
> Should the breakpoint audio cue also trigger every time the cursor column changes?

I don't think so, since the breakpoint applies to the line

**Comment by meganrogge** [comment]
> If the error cue should only play when the cursor is on an error decoration, the ErrorMarkerLineFeature needs to be adjusted too.

I think they both use `MarkerLineFeature`, which has been updated?

**Review by hediet — APPROVED** [review]

**Pull request merged**